src/Matrix/LabeledMatrix.php

```diff
@@ -21,34 +21,34 @@ class LabeledMatrix extends Matrix
 	 * @param object[] $colLabels
 	 * @throws Exception when either $rowLabels or $colLabels is empty
 	 */
-	public function __construct($rowLabels, $colLabels)
+	public function __construct(array $rowLabels, array $colLabels)
 	{
 		Assertions::assertLargerEqual(1, count($rowLabels), "Expected at least 1 row");
 		Assertions::assertLargerEqual(1, count($colLabels), "Expected at least 1 column");
 		$size = max(count($rowLabels), count($colLabels));
 		parent::__construct($size);
 
 		$this->rowLabels = new SplObjectStorage();
-		for ($i = 0; $i < count($rowLabels); $i++) {
-			$this->rowLabels[$rowLabels[$i]] = $i;
+		foreach ($rowLabels as $i => $label) {
+			$this->rowLabels[$label] = $i;
```

Comment:
More a note to myself than anything else, but this (old and new version) can result in errors when an associative array is passed in. The fix will likely be to add an assertion along the lines of `Assertions::assertArraySequential();` after line 27.